### PR TITLE
Improve program overview page.

### PR DESCRIPTION
**Pull request by mlehtima**
Improvements to program overview page and also added the ability to show program details from all list views through context menu the same way as from the first page.

### CODE REVIEW

**Comment by mlehtima** [comment]
This pull request is ready for review.

File: qml/js/yleApi.js [hunk 1]
```diff
@@ -73,6 +73,7 @@ var mapPrograms = function(programs) {
         return {
             id: program.id,
             title: program.title && program.title.fi,
```
Comment (joonne) [inline qml/js/yleApi.js:75]: at some point we need to make changes to this in order to support localization properly for the API content also. Just a side note :)

File: qml/pages/ProgramDelegate.qml [hunk 2]
```diff
@@ -3,10 +3,21 @@ import Sailfish.Silica 1.0
 
 import "../js/yleApi.js" as YleApi
 
-BackgroundItem {
+ListItem {
     id: listItem
-    width: ListView.view.width
-    height: Math.ceil(9*contentWidth/3/16)
+    contentWidth: ListView.view.width
+    contentHeight: Math.ceil(9*contentWidth/3/16)
```
Comment (joonne) [inline qml/pages/ProgramDelegate.qml]: I'd prefer parentheses & spaces for this kind of calculations for clarity:

`contentHeight: Math.ceil(((9 * contentWidth) / 3) / 16)`

not an issue any way, looks maybe a bit clearer.

Also, where does 3 & 16 come from, could they be constants?

File: qml/pages/ProgramDelegate.qml [hunk 3]
```diff
@@ -3,10 +3,21 @@ import Sailfish.Silica 1.0
...
+
+    menu: ContextMenu {
+        MenuItem {
+            text: qsTr("Show program info")
+            onClicked: {
```
Comment (joonne) [inline qml/pages/ProgramDelegate.qml]: block could be removed as it is not used?

File: qml/pages/ProgramOverviewPage.qml [hunk 4]
```diff
@@ -17,32 +17,70 @@ Page {
         anchors.fill: parent
         contentHeight: column.height
 
+        PullDownMenu {
```
Comment (joonne) [inline qml/pages/ProgramOverviewPage.qml]: this is cool :)

File: qml/pages/ProgramOverviewPage.qml [hunk 5]
```diff
@@ -17,32 +17,70 @@ Page {
...
+                        ? "http://images.cdn.yle.fi/image/upload/w_" + parent.width + ",h_" + Math.floor(parent.width * 16 / 9) + ",c_fit/" + program.image.id + ".jpg"
+                        : ""
+            }
+
+            Label {
+                id: itemTitle
+                color: Theme.primaryColor
+                text: program.itemTitle != program.title ? qsTr("Episode") + ": " + program.itemTitle : ""
+                truncationMode: TruncationMode.Fade
+                visible: text != ""
```
Comment (joonne) [inline qml/pages/ProgramOverviewPage.qml]: !==

File: qml/pages/ProgramOverviewPage.qml [hunk 6]
```diff
@@ -17,32 +17,70 @@ Page {
...
+                width: parent.width
+            }
+            Label {
+                id: episode
+                color: Theme.primaryColor
+                text: YleApi.formatProgramDetails(program.seasonNumber, program.episodeNumber)
+                truncationMode: TruncationMode.Fade
+                visible: text != ""
```
Comment (joonne) [inline qml/pages/ProgramOverviewPage.qml]: !==

File: qml/js/yleApi.js (hunk 1 above)
Comment (mlehtima) [reply qml/js/yleApi.js:75]: Yes that should be done at some point.

**Comment by mlehtima** [comment]
Changed code based on comments. Calculation of contentHeight in ProgramDelegate was indeed something really odd, I replaced it with a value from the theme.

**Comment by joonne** [comment]
looks good, I'll merge. Apparently GitHub offers squash & merge from the UI also :)

**Comment by mlehtima** [comment]
Actually the code was in two commits on purpose, those were changing different parts of the code.

**Comment by joonne** [comment]
sorry, my bad! :/ I did not pay attention enough then. Although I think I noticed that but I just got carried away with the automatic squash feature...

**Pull request merged**